src/rdb.c

```diff
@@ -2716,42 +2719,40 @@ void rdbLoadProgressCallback(rio *r, const void *buf, size_t len) {
     }
 }
 
-int rdbFunctionLoad(rio *rdb, int ver, functionsCtx* functions_ctx) {
+int rdbFunctionLoad(rio *rdb, int ver, functionsCtx* functions_ctx, sds *err) {
```

Comment:
let's add a comment saying that when it fails, error output is mandatory, and that releasing it is the responsibility of the caller.
another possibility is to avoid dynamic string, and just return static ones.

Reply:
It can not be a static string because it might be set by the engine itself (telling the user where the code failed compilation for example), will document it.